src/crc32.rs

```diff
@@ -21,6 +23,15 @@ impl Crc32 {
     pub fn new(crc: CRC) -> Self {
         let mut new = Self { periph: crc };
         new.init();
```

Comment:
You probably want to enable the clocks before this.

Reply:
Moved to be the first thing `Crc32::new` does. I don't believe it makes a considerable difference as calculations arn't done until the first write to the data register; which occurs outside this constructor.

Reply:
Thanks, the difference was that the `init` was resetting an internal state, which probably wouldn't do anything with the clocks disabled.

Reply:
The implementation of `init` sets the `RESET` bit of `CRC_CR`, which is automatically cleared by the hardware.
Even if the reset doesn't occur until after the clock is enabled, I suspect the device would still be reset.

Either way, its a fair point and I have no problem enabling the clock earlier.